controllers/pxc/perconaxtradbcluster_controller.go

```diff
@@ -0,0 +1,250 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PerconaXtraDBClusterReconciler reconciles a PerconaXtraDBCluster object
+type PerconaXtraDBClusterReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=pxc.percona.com,resources=perconaxtradbclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=pxc.percona.com,resources=perconaxtradbclusters/status,verbs=get;update;patch
+
+func (r *PerconaXtraDBClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("perconaxtradbcluster", req.NamespacedName)
+
+	rr := reconcile.Result{
+		RequeueAfter: time.Second * 5,
+	}
+
+	o := &pxcv1.PerconaXtraDBCluster{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, o)
+	if err != nil {
+		return rr, err
+	}
+
+	if _, ok := o.Annotations["percona.com/issue-vault-token"]; !ok {
+		return rr, nil
+	}
+
+	log.Info("found annotation")
+
+	opNs, err := operatorNamespace()
+	if err != nil {
+		return rr, errors.Wrap(err, "get operator namespace")
+	}
+
+	newSecretObj := corev1.Secret{}
+	err = r.Client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: o.Namespace,
+			Name:      o.Spec.VaultSecretName,
+		},
+		&newSecretObj,
+	)
+	if !apierrors.IsNotFound(err) {
+		return rr, err
+	}
+	if err == nil {
+		log.Info("issued secret was found, delete annotaton")
+		err = r.deleteAnnotation(o)
+		return rr, err
+	}
+
+	rootSecretName, err := rootSecretName()
+	if err != nil {
+		return rr, err
+	}
+
+	rootSecretObj := corev1.Secret{}
+	err = r.Client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: opNs,
+			Name:      rootSecretName,
+		},
+		&rootSecretObj,
+	)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("root secret was not found")
+			return rr, nil
+		}
+		return rr, err
+	}
+
+	err = r.IssueVaultToken(rootSecretObj, o.Spec.VaultSecretName, o.Namespace)
+	if err != nil {
+		return rr, err
+	}
+
+	log.Info("token was issued, delete annotation")
+
+	return rr, r.deleteAnnotation(o)
+}
+
+func (r *PerconaXtraDBClusterReconciler) deleteAnnotation(o *pxcv1.PerconaXtraDBCluster) error {
+	return r.Client.Patch(context.Background(), o, client.RawPatch(types.JSONPatchType, []byte("[{\"op\": \"remove\", \"path\": \"/metadata/annotations/percona.com~1issue-vault-token\"}]")))
+}
+
+func rootSecretName() (string, error) {
+	if s, ok := os.LookupEnv("VAULT_SECRET_NAME"); ok {
+		return s, nil
+	}
+	return "", errors.New("VAULT_SECRET_NAME env is not set")
+}
+
+func (r *PerconaXtraDBClusterReconciler) IssueVaultToken(rootVaultSercet corev1.Secret, newSecretName string, customerNamespace string) error {
+	data := string(rootVaultSercet.Data["keyring_vault.conf"])
+	fields := strings.Split(data, "\n")
+	conf := make(map[string]string)
+	for _, f := range fields {
+		kv := strings.Split(f, "=")
+		conf[strings.TrimSpace(kv[0])] = strings.TrimSpace(kv[1])
+	}
```

Comment:
please use `gopkg.in/ini.v1` - keyring config is in my.cnf format

Reply:
I tried to use it but as I remember it had some problem with this data. Not sure 100%, but maybe it did not like spaces or something.
I any case I'm not sure we should use entire lib for  such a simple task. What do you think?